addon/snapshot.js

```diff
@@ -57,6 +57,34 @@ function setTextareaContent(dom) {
   return dom;
 }
 
+// Copy attributes from Ember's rootElement to the DOM snapshot <body> tag. Some applications rely
+// on setting attributes on the Ember rootElement (for example, to drive dynamic per-route
+// styling). In tests these attributes are added to the #ember-testing container and would be lost
+// in the DOM hoisting below, so we copy them to the to the snapshot's <body> tag to
+// make sure that they persist in the DOM snapshot.
+function copyAttributesToBodyCopy(bodyCopy, testingContainer) {
+  let attributesToCopy = testingContainer.prop('attributes');
+  $.each(attributesToCopy, function() {
+    // Treat the class attribute as special - merge classes from the body and testingContainer
+    if(this.name == 'class') {
+      if(bodyCopy.attr('class') && this.value) {
+        let bodyClasses = bodyCopy.attr('class').split(' ');
+        let testingContainerClasses = this.value.split(' ');
+        testingContainerClasses.forEach(function(className) {
+          if(!bodyClasses.includes(className)) {
```

Comment:
I think I'm missing some subtlety here... Could we not take the classes of the testing container and the body and just concat them? What if there are two of the same classes on an element? I think it won't make a functional difference? But it would look messier.

So this would definitely make that nice and clean.

Reply:
Yeah...  Concat'ing them probably wouldn't present any problems - even if there were duplicates.  This just goes a little further to make sure we're creating duplicates.

If you'd strongly prefer the simpler concat version, for readability and maint benefits, I'd be simplify it.

Reply:
I'd definitely be in favor of simplifying this as much as we can.